base_business_document_import/models/business_document_import.py

```diff
@@ -69,7 +107,8 @@ def _match_partner(self, partner_dict, chatter_msg, partner_type="supplier"):
         if partner_dict.get("id"):
             return rpo.browse(partner_dict["id"])
         company_id = self._context.get("force_company") or self.env.user.company_id.id
-        domain = ["|", ("company_id", "=", False), ("company_id", "=", company_id)]
+        domain = domain or []
```

Comment:
`domain` should be copied before mutating, to prevent side effects on the domain uptream

Reply:
In fact, I'm relying on side effect as of here: https://github.com/OCA/edi/pull/236/files#diff-3bde8c0adf791caf40e4df6ec01a1e45fddd4298a4810398f4d559162960b95aR340
Any better way of doing this?